test/e2e/tests/network/add-custom-network.spec.js

```diff
@@ -113,12 +113,27 @@ const selectors = {
 
 async function navigateToAddNetwork(driver) {
   await driver.clickElement(selectors.accountOptionsMenuButton);
+  if (process.env.METAMASK_BUILD_TYPE === 'mmi') {
+    await checkIfPortfolioDashboardIsVisible(driver);
```

Comment:
The race-condition identified in this test can also affect other mmi tests: in all tests where we click the Settings icon `'[data-testid="account-options-menu-button"]',`.

For this reason, we could have a generic function for checking if settings is ready, before proceeding with the next action.

We don't need to do this in this PR, but we could do it in the Page Object implementation, and having this guard for mmi builds can mitigate all flakiness related to the settings re-render case 

cc @chloeYue

Reply:
Yes, I agree with you. We could do it in separate PR

Reply:
yeah agree ! POM will definitely help on it.

Reply:
@seaona, I have introduced an additional wait for the portfolio dashboard and removed the checkIfPortfolioDashboardIsVisible function. 

However, I believe this spec file is critical and contains many test cases that could be implemented using the Page Object Model. Do let me know what you think? 

Also, this is completely different discuss and will leave it to @chloeYue

Reply:
yeah agree, POM will definitely help on this !